bin/test-crawl

```diff
@@ -244,12 +246,104 @@ def get_referrers(url):
         ref_list.append(url)
     return ref_list
 
+nowtime = None
+nowstrn = None
+def nowstr():
+    global nowtime, nowstrn
+    if nowtime != time.time():
+        nowtime = time.time()
+        nowstrn = timezone.now().strftime('%H:%M:%S').encode()
+    return nowstrn
+
+b_exthost = re.sub(b'https?', b'', args.exthost.encode()) if args.exthost else None
+def normalize_for_diff(page):
+    # pages containing 'current time' can differ if they're fetched on different seconds:
+    #page = page.replace(nowstr(), b'00:00:00')
+    page = page.replace(b'https://', b'http://')
+    # regex replacements
+    page = re.sub(b'<!--.*?-->', b'', page)
+    page = re.sub(b' -- (Test|Development) Mode', b'', page)
+    page = re.sub(b'\n\s*\n+\s*', b'\n', page)
+    page = re.sub(b'name="csrfmiddlewaretoken" value="\w+"', b'', page)
+    page = re.sub(b'urn:uuid:[0-9a-f-]+', b'urn:uuid:00000000-0000-0000-0000-000000000000', page)
+    page = re.sub(b'<updated>[0-9T.:+-]+</updated>', b'', page)
+    page = re.sub(b'<published>[0-9T.:+-]+</published>', b'', page)
+    
+    if b_exthost:
+        page = re.sub(b_exthost, b'://testserver/', page)
+    return page
+
+def get_differences(a, b):
+    #open('a.html','wb').write(a)
+    #open('b.html','wb').write(b)
+    a = a.decode().splitlines()
+    b = b.decode().splitlines()
+    for group in difflib.SequenceMatcher(None,a,b).get_grouped_opcodes():
+        for tag, i1, i2, j1, j2 in group:
+            if tag == 'equal':
+                pass
+            elif tag == 'replace':
+                # see if the lines have next-day date matches
```

Comment:
Is this just to allow for the test run crossing over a day boundary?

Reply:
yes, though there's concern with it masking a date round difference in the long run. Let's keep it for now, but remember to try a run without it to see what differences it turns up.